src/cplscheme/ParallelCouplingScheme.cpp

```diff
@@ -25,26 +25,28 @@ ParallelCouplingScheme::ParallelCouplingScheme(
 
 void ParallelCouplingScheme::exchangeInitialData()
 {
+  bool initialDataExchange = true;
```

Comment:
We had a similar `bool` previously, but I think we need it again, if we want to properly support `substeps="false"`. Upside: We are closer to the v2 behavior and might also be able to remove some unnecessary communication of data at `WINDOW_START` for `substeps="true"`, if we have such a `bool` (Would suggest to take care of this outside of this PR).